src/components/FindMovie/FindMovie.js

```diff
@@ -1,55 +1,123 @@
-        </p>
-      </div>
+export const FindMovie = ({ movies, addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movieFromApi, setMovieFromApi] = useState('');
+  const [error, setError] = useState(false);
+  const [warn, setWarn] = useState(false);
+
+  const addMovieWithBenefits = () => {
+    setQuery('');
+    setMovieFromApi('');
+    addMovie([
+      ...movies,
+      {
+        title: movieFromApi.Title,
+        description: movieFromApi.Plot,
+        imgUrl: movieFromApi.Poster,
+        imdbUrl: `https://www.imdb.com/title/${movieFromApi.imdbID}/`,
+        imdbId: movieFromApi.imdbID,
+      },
+    ]);
+  };
+
+  return (
+    <>
+      <form
+        className="find-movie"
+        onSubmit={event => event.preventDefault()}
+      >
+        <div className="field">
+          <label className="label" htmlFor="movie-title">
+            Movie title
+          </label>
+
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className={`input${error || warn ? ' is-danger' : ''}`}
+              value={query}
+              onChange={(event) => {
+                setQuery(event.target.value);
+                setError(false);
+                setWarn(false);
+                setMovieFromApi('');
+              }}
+            />
+          </div>
+          {error ? (
+            <p className="help is-danger">
+              Can&apos;t find a movie with such a title
+            </p>
+          ) : null}
+          {warn ? (
+            <p className="help is-danger">
+              It&apos;s already in the list
+            </p>
+          ) : null}
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
         </div>
 
-        <div className="control">
-          <button
-            type="button"
-            className="button is-primary"
-          >
-            Add to the list
-          </button>
+        <div className="field is-grouped">
+          <div className="control">
+            <button
+              type="button"
+              className="button is-light"
+              onClick={() => {
+                getMovie(query)
+                  .then((result) => {
+                    setMovieFromApi({ ...result });
+                    setError(result.Response !== 'True');
+                  });
+              }}
+            >
+              Find a movie
+            </button>
+          </div>
+
+          <div className="control">
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={
+                () => (
+                  !movies.find(movie => movieFromApi.imdbID === movie.imdbId)
+                  && movieFromApi.imdbID
```

Comment:
```suggestion
                  !movies.some(movie => movieFromApi.imdbID === movie.imdbId)
```
доречніше використоти тут метод some, так як нас цікавить просто наявність фільму, а не сам фільм. 
І не зовсім розумію, нащо перевіряти movieFromApi.imdbID, воно в методі в тебе і так перевіряється, якшо не буде movieFromApi.imdbID, то метод видасть false

Reply:
за some +, тут ще перевіряю, чи не пусто, просто криво було)